prog/encoding_test.go

```diff
@@ -312,6 +312,20 @@ func TestDeserialize(t *testing.T) {
 			Out:       `test$opt2(0x0)`,
 			StrictErr: `non-nil argument for nil type`,
 		},
+		{
+			In:        `test$opt2(0x0) (non_existing_prop: 123)`,
```

Comment:
Please add a test with non-existent property followed by an existent property. It will test that parsing recovers after an unknown property.

Reply:
Hmm, a good idea. Added such a test.